pkg/segment/aggregations/evalaggs.go

```diff
@@ -323,93 +569,114 @@ func ComputeAggEvalForAvg(measureAgg *structs.MeasureAggregator, sstMap map[stri
 	return nil
 }
 
-func ComputeAggEvalForCardinality(measureAgg *structs.MeasureAggregator, sstMap map[string]*structs.SegStats, measureResults map[string]utils.CValueEnclosure, runningEvalStats map[string]interface{}) error {
+// Always pass valid strSet when using this function
+func PerformAggEvalForCardinality(measureAgg *structs.MeasureAggregator, strSet map[string]struct{}, fieldToValue map[string]utils.CValueEnclosure) (float64, error) {
```

Comment:
> // Always pass valid strSet when using this function

What do you mean by `valid`? Do you mean anything that's not nil, or something more specific?

Reply:
Yes, I mean that is not nil. Will update it.